fish-rust/src/builtins/shared.rs

```diff
@@ -114,6 +114,12 @@ impl output_stream_t {
         self.ffi().append(&s.as_ref().into_cpp())
     }
 
+    /// Append a &wstr or WString with a newline
+    pub fn appendln(&mut self, s: impl Into<WString>) -> bool {
```

Comment:
I think this trait-bound should ideally be `impl ToWString`, but afaik we only really implement that for some integer-types, and not for `T where T: ToString`, or some other very generic impl.

Note that any type that `impl Display` automatically `impl ToString` per the [docs](https://doc.rust-lang.org/std/fmt/trait.Display.html) .

TBH I'm probably rather of the opinion we should just remove `ToWString` and instead use `.to_string()` and convert to UTF32, unless there is some behaviour-difference I am not aware of